File: Python/LRU Cache.py
class LRUCache:
    # @param capacity, an integer
    def __init__(self, capacity):
        self.cache = {}
        self.capacity = capacity
        self.queue = []
        

    # @return an integer
    def get(self, key):
        if key in self.cache:
            self.updateQueue(key)
            return self.cache[key]
        else:
            return -1
        

    # @param key, an integer
    # @param value, an integer
    # @return nothing
    def set(self, key, value):
        if key in self.cache:
            self.cache[key] = value
            self.updateQueue(key)
        else:
            if len(self.queue) == self.capacity:
                del self.cache[self.queue.pop(0)]
            self.cache[key] = value
            self.queue.append(key)
        
    def updateQueue(self, key):
        self.queue.remove(key)
        self.queue.append(key)

File: Java/src/LRU Cache.java
public class LRUCache {
    private final int capacity;
    private final Map<Integer, Integer> cache;
    private final List<Integer> queue; 

    public LRUCache(int capacity) {
        this.capacity = capacity;
        this.cache = new HashMap<Integer, Integer>();
        this.queue = new ArrayList<Integer>();
    }
    
    private void updatequeue(int key){
    	queue.remove(queue.indexOf(key));
    	queue.add(key);
    }

    public int get(int key) {
        if (!cache.containsKey(key)) {
            return -1;
        }
       updatequeue(key);
       return cache.get(key);
    }

    public void set(int key, int value) {
        int val = get(key); 
        cache.put(key, value); 
        if (val == -1) {
            queue.add(key); 

            if (cache.size() > capacity) { 
                cache.remove(queue.get(0));
                queue.remove(0);
            }
        }
    }
}